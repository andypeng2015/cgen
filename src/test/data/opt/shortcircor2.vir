module foo

export function w $foo(l %a, l %i) {
@start:
  %k = sge.l %i, 3_l
  %f = flag.w %k
  %c = ne.w %f, 0_w
  br %c, @join, @cont
@cont:
  %k = slt.l %i, 0_l
  %f = flag.w %k
  jmp @join
@join:
  %c = ne.w %f, 0_w
  br %c, @bad, @ok
@ok:
  %s = mul.l %i, 4_l
  %a = add.l %a, %s
  %v = ld.w %a
  ret %v
@bad:
  ret 0xffffffff_w
}
