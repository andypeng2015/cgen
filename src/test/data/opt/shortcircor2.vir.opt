module foo

export function w $foo(l %a, l %i) {
@2:
  %1 = sge.l %i, 0x3_l ; @17
  br %1, @16(0xffffffff_w), @4
@4:
  %5 = slt.l %i, 0x0_l ; @21
  br %5, @16(0xffffffff_w), @11
@11:
  %3 = lsl.l %i, 0x2_l ; @19
  %4 = add.l %a, %3 ; @20
  %v.1 = ld.w %4 ; @15
  jmp @16(%v.1)
@16(%0):
  ret %0
}
