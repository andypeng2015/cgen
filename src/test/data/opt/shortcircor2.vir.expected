module foo

export function w $foo(l %a, l %i) {
@0:
  %1 = sge.l %i, 0x3_l ; @15
  br %1, @14(0xffffffff_w), @2
@2:
  %3 = slt.l %i, 0x0_l ; @17
  %4 = flag.w %3 ; @18
  %5 = ne.w %4, 0x0_w ; @19
  br %5, @14(0xffffffff_w), @9
@9:
  %6 = lsl.l %i, 0x2_l ; @20
  %7 = add.l %a, %6 ; @21
  %v.1 = ld.w %7 ; @13
  jmp @14(%v.1)
@14(%0):
  ret %0
}
